weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts

```diff
@@ -98,43 +98,96 @@ const useCalls = (
   opts?: {skip?: boolean}
 ): Loadable<CallSchema[]> => {
   const getTsClient = useGetTraceServerClientContext();
-  const loadingRef = useRef(false);
-  const [callRes, setCallRes] =
-    useState<traceServerClient.TraceCallsQueryRes | null>(null);
+  // if allCallsLoaded is true, we have loaded all calls, this is how we tell if we are loading
+  // if skip is true, we don't need to load any calls
+  const [allCallsLoaded, setAllCallsLoaded] = useState(opts?.skip ?? false);
+  const [callRes, setCallRes] = useState<traceServerClient.TraceCallSchema[]>(
+    []
+  );
   const deepFilter = useDeepMemo(filter);
+
+  // if a limit is provided, we use that and only return the single page
+  // otherwise we use a default of 10,000 and load 50 pages to return all calls
+  // Arbitrary page limit of 10,000, in conjunction with the max 50 pages limits calls to 500,000
+  const pageSize = limit ?? 10000;
```

Comment:
I think you might have an issue if the user passes a limit > 10_000

Reply:
I just figured if they really wanted a limit of over 10k , they can get it.
Would i be better to have some mechanism to get the exact amount of the limit if limit is over 10k?
eg limit of 47k
get 10 + 10 + 10 + 10 + 7?

Reply:
gotcha i got that working

https://github.com/wandb/weave/assets/25037002/b98e0802-3e98-4d69-85fe-175cbcf7ace0